controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java

```diff
@@ -429,6 +436,17 @@ public LongStream getLogChunkIds(ApplicationId id, JobType type) {
                       .sorted();
     }
 
+    // -------------- Audit log -----------------------------------------------
+
+    public AuditLog readAuditLog() {
+        return readSlime(auditLogPath()).map(auditLogSerializer::fromSlime)
+                                        .orElse(AuditLog.empty);
+    }
+
+    public void writeAuditLog(AuditLog log) {
```

Comment:
Should this take the lock as argument such that nobody accidentally forgets to lock? Or at least javadoc it ...

Reply:
Considered, but `AuditLogger` should really remain the only writer...

And we don't do it for the similar cases in this class, instead we try to enforce such requirements one level up. E.g. `ApplicationController#store(LockedApplication)` or `RotationRepository#getOrAssignRotation(Application, RotationLock)`.